dig.go

```diff
@@ -60,24 +60,19 @@ type InvokeOption interface {
 
 // Container is a directed acyclic graph of types and their dependencies.
 type Container struct {
-	nodes map[key]*node
-	cache map[key]reflect.Value
+	// Mapping from key to all the nodes that can produce a value for that
+	// key.
+	producers map[key][]*node
```

Comment:
Is `providers` a more fitting name here? It might improve readability to make "provider" and "producer" the same or obviously different.

Reply:
I like providers. SG